lib/Importer/ONNXModelLoader.cpp

```diff
@@ -1459,13 +1469,35 @@ Error ONNXModelLoader::loadPool(const ONNX_NAMESPACE::NodeProto &op,
   // Load the inputs:
   NodeValue in;
   ASSIGN_VALUE_OR_RETURN_ERR(in, getNodeValueByName(op.input(0)));
+
+  std::vector<unsigned_t> kernelsShape;
+  ASSIGN_VALUE_OR_RETURN_ERR(kernelsShape,
+                             getShape<unsigned_t>(dict["kernel_shape"]));
+
   std::vector<unsigned_t> strides(2, 1);
-  if (dict.count("strides")) {
-    ASSIGN_VALUE_OR_RETURN_ERR(strides, getShape<unsigned_t>(dict["strides"]));
+
+  size_t inDim = in.dims().size();
+  size_t kerDim = kernelsShape.size();
+
+  std::vector<unsigned_t> kernels = {1, kernelsShape[kerDim - 1]};
+
+  // For maxPool1D inDim = 3
+  if (inDim == 3) {
+    in = G_->createExpandDims(opName, in, 2);
+  }
+
+  if (kerDim == 2) {
+    kernels[0] = kernelsShape[0];
+    if (dict.count("strides")) {
+      ASSIGN_VALUE_OR_RETURN_ERR(strides,
+                                 getShape<unsigned_t>(dict["strides"]));
+    }
+  } else if (kerDim == 1) { // For maxPool1D
+    if (dict.count("strides")) {
+      strides[1] = dict.at("strides")->ints(0);
+      strides[0] = 1;
+    }
```

Comment:
In order to make this cleaner and less error prone, it feels like we should handle this in two cases: one for maxpool 1D and one for 2D. I.e. we should check one of these conditions (e.g.  `if (inDim == 3)`) and then in there return an error if `kerDims != 1` and setup strides correctly. Otherwise we check that the dimensions make sense for maxpool 2D. Otherwise we could have some weird situation here e.g. where we have a malformed maxpool with `inDim == 3 && kerDim == 2` and we wouldn't recognize it.

Reply:
Done.